src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -154,50 +161,104 @@ void VehicleInfoPlugin::UnsubscribeFromRemovedVDItems() {
   application_manager_->GetRPCService().ManageHMICommand(message);
 }
 
-bool IsOtherAppAlreadySubscribedFor(
-    const std::string& ivi_name,
-    const application_manager::ApplicationManager& app_mngr,
-    const uint32_t current_app_id) {
-  auto applications = app_mngr.applications();
+void VehicleInfoPlugin::ProcessResumptionSubscription(
+    application_manager::Application& app,
+    VehicleInfoAppExtension& ext,
+    resumption::Subscriber subscriber) {
+  LOG4CXX_AUTO_TRACE(logger_);
 
-  for (auto& app : applications.GetData()) {
-    auto& ext = VehicleInfoAppExtension::ExtractVIExtension(*app);
-    if (ext.isSubscribedToVehicleInfo(ivi_name) &&
-        (app->app_id() != current_app_id)) {
-      return true;
+  auto pending = ext.PendingSubscriptions().GetData();
+  for (const auto& ivi : pending) {
+    if (IsSubscribedAppExist(ivi)) {
```

Comment:
why only sub if another app is subbed? I don't see this in previous logic

Reply:
@iCollin it just means that SDL should not wait for pending resumption for some specific VD if it is already successfully subscribed by another application. In that case SDL can safely remove pending resumption for such VD and mark extension as subscribed immediately.
If there is no another apps subbed to specific VD or subscription is in progress (pending) then we can't mark extension as subscribed right away.
This is a tricky logic because SDL should send SubscribeVD request to HMI only when first application subscribes to specific VD and when last application unsubscribes from specific VD. All other applications which are going to subscribe to the same VD should not trigger sending of SubscribeVD request to HMI